### PR TITLE
Rename `ensure_default_get` and add test

### LINKED ISSUE #5576: Rename `ensure_default_get`
We have a utility function currently named `ensure_default_get`

https://github.com/dask/distributed/blob/4cc0b0618ff4aa6c29643da3bc9d460630aa8556/distributed/client.py#L4750-L4752

however, our usage of `get` has generally been replaced by `scheduler` or `client` throughout the codebase. We might consider renaming `ensure_default_get` to `ensure_default_client` to better match the `default_client()` method and be more consistent with our naming standard. 

Additionally, I noticed `ensure_default_get` doesn't have any tests. We should also add a test to ensure that this utility works as expected.

### PR DESCRIPTION
**Pull request by ncclementi**
- [x] Closes #5576
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by ncclementi** [comment]
There are multiple CI failures but all seem unrelated.

cc: @jrbourbeau for review

File: distributed/client.py [hunk 1]
```diff
@@ -4741,7 +4741,8 @@ def default_client(c=None):
         )
 
 
-def ensure_default_get(client):
+def ensure_default_client(client):
+    "Ensures the client passed as argument is set as the default."
```
Comment (jrbourbeau) [inline distributed/client.py:4745]: Style nit: Single and triple-quote Python strings have slightly different behavior. Most folks use triple quotes everywhere for docstrings

```suggestion
    """Ensures the client passed as argument is set as the default"""
```

File: distributed/client.py [hunk 2]
```diff
@@ -4741,7 +4741,8 @@ def default_client(c=None):
         )
 
 
-def ensure_default_get(client):
+def ensure_default_client(client):
```
Comment (jrbourbeau) [inline distributed/client.py:4744]: This name change will break user's code who are using the old name today. To make this name change process smoother, let's make sure the old name still works, but raises a deprecation warning asking users to switch to the new name. 

One nice way to accomplish this is to add a [module-level `__getattr__` function](https://www.python.org/dev/peps/pep-0562/) to the bottom of this file. Something like the following should do the trick (though I haven't tested any of the code below)

```python
def __getattr__(name):
    if name == "ensure_default_get":
        use_instead = _deprecations[name]

        warnings.warn(
            f"`ensure_default_get` is deprecated and will be removed in a future release. "
            f"Please use `distributed.client.ensure_default_client` instead.",
            category=FutureWarning,
            stacklevel=2,
        )
        return ensure_default_client
    raise AttributeError(f"module {__name__} has no attribute {name}")
```

Comment (jrbourbeau) [reply distributed/client.py:4744]: We can also add a test similar to 

https://github.com/dask/distributed/blob/1bbb7f7474e27a434e97427c518f7fdd8e255dd5/distributed/tests/test_utils.py#L620-L623

to checks that when one tries to import `ensure_default_get` that an informative warning is raised and `ensure_default_client` is returned instead

**Review by jrbourbeau — COMMENTED** [review]
Thanks @ncclementi! This looks great

**Review by jrbourbeau — APPROVED** [review]
Thanks for the updates @ncclementi! Test failures are unrelated -- this is in

**Pull request merged**